src/main/java/mate/academy/service/impl/UserServiceImpl.java

```
@@ -0,0 +1,35 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        try {
+            return userDao.add(user);
+        } catch (DataProcessingException e) {
+            System.out.println("Can't add user to db" + user);
+            return user;
+        }
```

This violates checklist item #10: 'You should create salt and hash password in UserService add() method.' The responsibility for creating the salt and hashing the password belongs here. The current implementation just delegates to the DAO without preparing the user object correctly.

Additionally, the error handling is incorrect. Returning the user object in the `catch` block hides the failure from the calling service. If `userDao.add(user)` fails, you should re-throw the `DataProcessingException`.